### PR TITLE
Optest: Allow parametrized names for xfails checks

### PR DESCRIPTION
**Pull request by NicolasHug**
CC @zou3519 

This is hopefully a fix for https://github.com/pytorch/vision/pull/8058/files#r1368570541. It seems to work for me locally, but maybe there's a more elegant way of handling this?

### CODE REVIEW

File: torch/testing/_internal/optests/generate_tests.py [hunk 1]
```diff
@@ -333,6 +333,7 @@ def validate_failures_dict_structure(
                 if not actual_test_name.startswith(test):
                     continue
                 base_test_name = actual_test_name[len(test) + 2 :]
+                base_test_name = base_test_name.split("[")[0]  # remove potential pytest parametrization suffix
```
[the remark this answers is not in the document]
Comment (NicolasHug) [reply torch/testing/_internal/optests/generate_tests.py:336]: > a way to detect if we're using pytest 

The [documented way ](https://pytest.org/en/6.2.x/example/simple.html#detect-if-running-from-within-a-pytest-run) is to create a global variable on the module that gets set to True in `conftest.py`. There are other suggested ways [on SO](https://stackoverflow.com/questions/25188119/test-if-code-is-executed-from-within-a-py-test-session), but they may not all apply here.

Is it worth it though? Brackets can't be part of test names anyway, so if there are brackets, they *must* come from pytest?

[the remark this answers is not in the document]
Comment (NicolasHug) [reply torch/testing/_internal/optests/generate_tests.py]: > Also, the full regex on the two brackets "test_name[.*]" would be more robust.

done!

File: torch/testing/_internal/optests/generate_tests.py [hunk 2]
```diff
@@ -333,6 +334,8 @@ def validate_failures_dict_structure(
                 if not actual_test_name.startswith(test):
                     continue
                 base_test_name = actual_test_name[len(test) + 2 :]
+                # remove potential pytest parametrization suffix
+                base_test_name = re.sub("\[.*\]", "", base_test_name)
```
Comment (NicolasHug) [inline torch/testing/_internal/optests/generate_tests.py:338]:
```suggestion
                base_test_name = re.sub(r"\[.*\]", "", base_test_name)
```

**Review by zou3519 — APPROVED** [review]

**Pull request closed without merging**